### PR TITLE
Support for which-key

### PR DESCRIPTION
**Pull request by maxsandova**
The key "?" can be changed to "h" since it was its default, it will be invoked after "C-h h". "g?" won't work here because of the way which-key is coded. The other possible bindings are commented out.

### CODE REVIEW

File: evil-collection-which-key.el [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+;;; Code:
+
+(require 'evil)
+(require 'which-key nil t)
+
+;; which key is coded so that the promp properly shows j and k as the bindings
+(defun evil-collection-which-key-setup ()
+  "Set up `evil' bindings for `which-key'."
+  
+  ;; (define-key which-key-C-h-map "a" 'which-key-abort)
```
Comment (jojojames) [inline evil-collection-which-key.el]: What about 'q' for abort?

Comment (maxsandova) [reply evil-collection-which-key.el]: Oh, that was a no-brainer how I did not come up with that...

File: evil-collection-which-key.el [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+;;; Code:
+
+(require 'evil)
+(require 'which-key nil t)
+
+;; which key is coded so that the promp properly shows j and k as the bindings
+(defun evil-collection-which-key-setup ()
+  "Set up `evil' bindings for `which-key'."
+  
+  ;; (define-key which-key-C-h-map "u" 'which-key-undo-key)
```
Comment (jojojames) [inline evil-collection-which-key.el]: I bet this is already defined but can you explicitly define it here? (e.g. remove the ;;)

**Comment by jojojames** [comment]
This is almost ready to be merged, can you add a 'defvar' on the keymap though?

```
evil-collection-which-key.el:39:15:Error: reference to free variable ‘which-key-C-h-map’
```

You can run 'make' to see if your code byte compiles cleanly.

**Comment by jojojames** [comment]
Thanks for follow-through!

**Comment by maxsandova** [comment]
Could you be more specific on how should I use make?  I installed cask and ran make,  I basically got 
`evil-collection-ag.el:31:1:Error: Cannot open load file: No such file or directory, evil` 
all over the place

**Comment by jojojames** [comment]
Did you install the cask dependencies with 'cask'?

You should have a .cask folder like this.

```
  drwxr-xr-x  3 james staff   102 11-15 18:50  .cask
  drwxr-xr-x  3 james staff  102 11-15 18:50 25.3
  drwxr-xr-x 15 james staff 510 12-06 17:49 elpa
  drwxr-xr-x  6 james staff  204 11-20 23:13 ansi-20150703.126
  drwxr-xr-x  4 james staff  136 11-20 23:13 archives
  drwxr-xr-x  6 james staff  204 11-20 23:13 commander-20140120.1052
  drwxr-xr-x  6 james staff  204 11-20 23:13 dash-20171028.854
  drwxr-xr-x 10 james staff  340 11-20 23:13 ert-runner-20171111.451
  drwxr-xr-x 41 james staff 1394 12-06 17:49 evil-20171129.651
  drwxr-xr-x  6 james staff  204 11-20 23:13 f-20171119.723
  drwx------ 10 james staff  340 12-06 17:49 gnupg
  drwxr-xr-x  6 james staff  204 11-15 18:50 goto-chg-20170917.1200
  drwxr-xr-x  6 james staff  204 12-06 17:49 package-lint-20171201.1903
  drwxr-xr-x  6 james staff  204 11-20 23:13 s-20171102.227
  drwxr-xr-x  6 james staff  204 11-20 23:13 shut-up-20150423.522
  drwxr-xr-x  6 james staff  204 11-15 18:50 undo-tree-20170706.246
```

**Pull request merged**